### PR TITLE
42 Invite controllers approve & reject backend routes

### PR DESCRIPTION
**Pull request by peacebonez**

### CODE REVIEW

File: routes/invitation.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  } catch (err) {
+    console.error(err.message);
+    res.status(500).send("Server Error");
+  }
+});
+
+//PUT request to reject invitation
+router.put("/:id/reject", async (req, res) => {
+  try {
+    try {
```
Comment (ThinkTink) [inline routes/invitation.js]: We don't need this double nested - just use one like in the above route

File: routes/invitation.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      res.status(404).send("Invitation not found");
+    }
+
+    invitation.status = "approved";
+
+    await invitation.save();
+
+    res.json(invitation);
+  } catch (err) {
+    console.error(err.message);
```
Comment (ThinkTink) [inline routes/invitation.js:26]:
```suggestion
```

File: routes/invitation.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+//PUT request to approve invitation
+router.put("/:id/approve", async (req, res) => {
+  try {
+    const invitation = await Invitation.findById(req.params.id);
+
+    if (!invitation) {
+      res.status(404).send("Invitation not found");
+    }
+
+    invitation.status = "approved";
```
Comment (ThinkTink) [inline routes/invitation.js]: Would we want to check the current status first? Like if an invite is marked as rejected, would we want to be able to reverse that and mark it as accepted?

File: routes/invitation.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      res.json(invitation);
+    } catch (err) {
+      console.error(err.message);
+      res.status(500).send("Server Error");
+    }
+  } catch (err) {
+    console.error(err.message);
+    res.status(500).send("Server Error");
+  }
+});
```
Comment (ThinkTink) [inline routes/invitation.js:54]: An invitation should only be approved or rejected by the person it was sent to right? So we'll want to make this an authenticated route and check that the user making the call is the same as the toEmail on the invite.

**Review by ThinkTink — COMMENTED** [review]
If all the front end files are replicated on your other branch, you can delete them from this one.
Also, I changed my mind about merging before auth is done - auth should be done soon so let's hold off on merging things that need to use protected routes until that's done first

File: routes/invitation.js (hunk 3 above)
Comment (peacebonez) [reply routes/invitation.js]: perhaps check if status === "pending" and that's the only circumstance we can modify the status?

Comment (peacebonez) [reply routes/invitation.js]: better yet... 
`  if (invitation.status !== "pending") {
      res
        .status(400)
        .json({ msg: `Invitation status already ${invitation.status}` });
    }
    invitation.status = "rejected";
    await invitation.save();`

**Comment by ThinkTink** [comment]
@peacebonez Let me know when this is ready for re-review

**Pull request merged**